services/app-web/src/pages/StateSubmission/RateDetails/V2/RateDetailsV2.tsx

```diff
@@ -266,8 +267,14 @@ const RateDetailsV2 = ({
             (options.type === 'CONTINUE' || options.type === 'SAVE_AS_DRAFT')
         ) {
             try {
-                const updatedRates = generateUpdatedRates(rateForms)
-
+                const formattedRateForms = rateForms.filter((rate) => {
+                    if (rate.ratePreviouslySubmitted === 'YES') {
+                        return rate.id
+                    } else {
+                        return rate
+                    }
```

Comment:
With `filter` I do need to return something after the if statement otherwise it will error

Reply:
I was also wondering what the purpose of this filter was since `generateUpdatedRates` seems to be similarly checking `ratePreviouslySubmitted`.

Reply:
Sorry just seeing this. I think the difference with `generateUpdatedRates` is that is uses `map` so everything in the original array is returned

Reply:
Oh, I understand what's going on. I was confused because for filter the callback function wants a truthy or falsy return value and my assumption was `rate.id` was always going to be defined. But in this instance, a linked rate without linking a rate would have no `rate.id` thus returning an undefined falsy value and filtering out that incomplete linked rate. If we did not do this then `updateDraftContractRates` would error because a undefined would be given as the `rateID`.

Does that sound about right?